pkg/kubectl/cmd/delete/delete.go

```diff
@@ -158,6 +159,12 @@ func (o *DeleteOptions) Complete(f cmdutil.Factory, args []string, cmd *cobra.Co
 			o.IgnoreNotFound = true
 		}
 	}
+
+	// If specifying to delete all namespaces, must enter a --grace-period flag
+	if (sets.NewString(args...).Has("ns") || sets.NewString(args...).Has("namespace")) && o.DeleteAll && o.GracePeriod == -1 {
```

Comment:
This change could violate forward compatibility. The users chose to delete all namespaces will find out the broken change. We need more discussion about whether to merge this or find a better approach.

Reply:
This is definitely true, I think the PR was more to get the dialog started (and for me to learn about the kubectl code base by working on some old issues...)

I was thinking about alternate approaches but couldn't find one right now. We could make prompt to stdin, but this would break scripts using `kubectl delete ns --all` as well. A warning would be superfluous too because the namespaces would be deleted in any event.

We could also leave as-is, which I definitely respect as an option. Most other apis, however, say something before going "nuclear", e.g. delete all namespaces.

I'm open to any other suggestions, however.